### PR TITLE
Skip wrapping nodes when the only match is an editor

### PR DESCRIPTION
**Pull request by TheSpyder**
**Description**
`Transforms.wrapNodes` attempts to detect when zero nodes match, but it missed an edge case where one node matches - the editor. The function very quickly throws `Error: Cannot get the parent path of the root path []` when this happens.

**Example**
The test shows the failure condition. The editor I'm working on supports `contenteditable=false` sections in document content, and while we are able to handle most of the filtering outside of slate this is one case that is difficult to handle without a lot of extra code.

**Context**
There are a few other ways I can think of to fix this - such as wrapping the supplied `match` function to enforce `!Editor.isEditor` - but this seemed like the least invasive way to deal with it.

**Checks**
- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)
- [x] You've [added a changeset](https://github.com/atlassian/changesets/blob/master/docs/adding-a-changeset.md) if changing functionality. (Add one with `yarn changeset add`.)

### CODE REVIEW

File: packages/slate/src/transforms/node.ts [hunk 1]
```diff
@@ -940,8 +940,14 @@ export const NodeTransforms: NodeTransforms = {
         if (matches.length > 0) {
           const [first] = matches
           const last = matches[matches.length - 1]
-          const [, firstPath] = first
-          const [, lastPath] = last
+          const [firstNode, firstPath] = first
+          const [lastNode, lastPath] = last
+
+          if (firstNode === lastNode && Editor.isEditor(firstNode)) {
```
Comment (spocke) [inline packages/slate/src/transforms/node.ts:946]: Wouldn't this just be `if (firstPath.length === 0 && lastPath.length === 0)`? Since the only way you would ever get editor in a nodePath pair is if the path is an empty path. Would give a slight performance boost since isEditor checks nodes etc. Would be more hard to read though so maybe a comment would make it clearer.

**Review by spocke — APPROVED** [review]

**Review by dylans — APPROVED** [review]
